src/chatClient.h

```diff
@@ -199,28 +199,39 @@ class PeerChatRoom: public ChatRoom
     virtual bool syncWithApi(const mega::MegaTextChat& chat);
     bool syncPeerPriv(chatd::Priv priv);
     static uint64_t getSdkRoomPeer(const ::mega::MegaTextChat& chat);
+    static chatd::Priv getSdkRoomPeerPriv(const ::mega::MegaTextChat& chat);
     void initWithChatd();
     virtual void connect();
+    UserAttrCache::Handle mUsernameAttrCbId;
     void updateTitle(const std::string& title);
     friend class Contact;
     friend class ChatRoomList;
     PeerChatRoom(ChatRoomList& parent, const uint64_t& chatid,
             unsigned char shard, chatd::Priv ownPriv, const uint64_t& peer,
             chatd::Priv peerPriv, uint32_t ts);
-    PeerChatRoom(ChatRoomList& parent, const mega::MegaTextChat& room, Contact& contact);
+    PeerChatRoom(ChatRoomList& parent, const mega::MegaTextChat& room);
     ~PeerChatRoom();
-    //@endcond
+
 public:
     virtual IApp::IChatListItem* roomGui() { return mRoomGui; }
     /** @brief The userid of the other person in the 1on1 chat */
     const uint64_t peer() const { return mPeer; }
     chatd::Priv peerPrivilege() const { return mPeerPriv; }
-    /** @brief The contact object representing the peer of the 1on1 chat */
-    Contact& contact() const { return mContact; }
+
+    /**
+     * @brief The contact object representing the peer of the 1on1 chat.
+     * @note Returns nullptr when the 1on1 chat is with a user who canceled the account
+     */
+    Contact *contact() const { return mContact; }
 
     /** @brief The screen name of the peer */
     virtual const std::string& titleString() const;
 
+    /** @brief The screen email address of the peer */
```

Comment:
You can prevent duplicating the email property in the contact and 1on1 room by selectively returning here mContact->mEmail or "Deleted user"